rust/signer/src/lib.rs

```diff
@@ -14,124 +14,116 @@
+fn update_seed_names(seed_names: &str) {
+    navigator::update_seed_names(seed_names)
+}
 
-    @Java_io_parity_signer_models_SignerDataModel_historyAcknowledgeWarnings
-    fn acknowledge_warnings(
-        dbname: &str
-    ) -> anyhow::Result<(), anyhow::Error> {
-        db_handling::manage_history::reset_danger_status_to_safe(dbname).map_err(|e| e.anyhow())
-    }
+fn qrparser_get_packets_total(data: &str, cleaned: bool) -> anyhow::Result<u32, ErrorDisplayed> {
+    qr_reader_phone::get_length(data, cleaned).map_err(Into::into)
+}
 
-    @Java_io_parity_signer_models_SignerDataModel_historyHistoryEntrySystem
-    fn history_entry_system(
-        entry: &str,
-        dbname: &str
-    ) -> anyhow::Result<(), anyhow::Error> {
-        db_handling::manage_history::history_entry_system(dbname, entry.to_string()).map_err(|e| e.anyhow())
-    }
+fn qrparser_try_decode_qr_sequence(
+    data: &str,
+    cleaned: bool,
+) -> anyhow::Result<String, anyhow::Error> {
+    qr_reader_phone::decode_sequence(data, cleaned)
+}
 
-    @Java_io_parity_signer_models_SignerDataModel_historySeedNameWasShown
-    fn seed_name_was_shown(
-        seed_name: &str,
-        dbname: &str
-    ) -> anyhow::Result<(), anyhow::Error> {
-        db_handling::manage_history::seed_name_was_shown(dbname, seed_name.to_string()).map_err(|e| e.anyhow())
-    }
+fn substrate_path_check(seed_name: &str, path: &str, network: &str, dbname: &str) -> String {
+    db_handling::interface_signer::dynamic_path_check(dbname, seed_name, path, network)
+}
 
-    @Java_io_parity_signer_models_SignerDataModel_testGetAllTXCards
-    fn get_all_tx_cards() -> String {
-        if let transaction_parsing::Action::Read(content) = transaction_parsing::test_all_cards::make_all_cards() {
-            format!("{{{}}}", content)
-        } else {
-            "".to_string()
-        }
-    }
+fn history_init_history_with_cert(dbname: &str) -> anyhow::Result<(), anyhow::Error> {
+    db_handling::cold_default::signer_init_with_cert(dbname).map_err(|e| e.anyhow())
+}
+
+fn history_init_history_no_cert(dbname: &str) -> anyhow::Result<(), anyhow::Error> {
+    db_handling::cold_default::signer_init_no_cert(dbname).map_err(|e| e.anyhow())
+}
+
+fn history_device_was_online(dbname: &str) -> anyhow::Result<(), anyhow::Error> {
+    db_handling::manage_history::device_was_online(dbname).map_err(|e| e.anyhow())
+}
 
-    @Java_io_parity_signer_models_SignerDataModel_testGetAllLogCards
-    fn get_all_log_cards() -> String {
-        definitions::history::print_all_events()
+fn history_get_warnings(dbname: &str) -> anyhow::Result<bool, anyhow::Error> {
+    db_handling::helpers::get_danger_status(dbname).map_err(|e| e.anyhow())
+}
+
+fn history_acknowledge_warnings(dbname: &str) -> anyhow::Result<(), anyhow::Error> {
+    db_handling::manage_history::reset_danger_status_to_safe(dbname).map_err(|e| e.anyhow())
+}
+
+fn history_entry_system(event: Event, dbname: &str) -> anyhow::Result<(), anyhow::Error> {
```

Comment:
This is supposed to allow UI to write generic strings into history. Should accept string and convert it to "system entry" event inside. The only usage now is logging "logging errors", not sure if it is useful at all, but let's just keep for now.